### PR TITLE
Ncp: Update the RESET_hanlder() to ensure a single spinel reset message goes out.

### PR DESCRIPTION
**Pull request by abtink**

### CODE REVIEW

**Comment by mszczodrak** [comment]
LGTM :+1:

File: src/ncp/ncp_base.cpp [hunk 1]
```diff
@@ -1022,11 +1024,17 @@ ThreadError NcpBase::CommandHandler_RESET(uint8_t header, unsigned int command,
...
-    mChangedFlags |= NCP_PLAT_RESET_REASON;
     otDisable();
-    mUpdateChangedPropsTask.Post();
 
-    return SendLastStatus(SPINEL_HEADER_FLAG | SPINEL_HEADER_IID_0, SPINEL_STATUS_RESET_SOFTWARE);
+    errorCode = SendLastStatus(SPINEL_HEADER_FLAG | SPINEL_HEADER_IID_0, SPINEL_STATUS_RESET_SOFTWARE);
+
+    if (errorCode != kThreadError_None)
+    {
+        mChangedFlags |= NCP_PLAT_RESET_REASON;
```
Comment (arjunasiva) [inline src/ncp/ncp_base.cpp]: setting mChangedFlags is now after `otDisable()`, no harm in that right?

**Comment by arjunasiva** [comment]
LGTM 👍

File: src/ncp/ncp_base.cpp (hunk 1 above)
Comment (abtink) [inline src/ncp/ncp_base.cpp]: Correct. It should not matter.

**Comment by jwhui** [comment]
LGTM 👍

**Pull request merged**